rules-tests/CodeQuality/Rector/Class_/AddReturnDocBlockToCollectionPropertyGetterByToManyAnnotationRector/Fixture/OneToMany/with_index_by.php.inc

```diff
@@ -0,0 +1,62 @@
+<?php
+
+namespace Rector\Doctrine\Tests\CodeQuality\Rector\Class_\AddReturnDocBlockToCollectionPropertyGetterByToManyAnnotationRector\Fixture\OneToMany;
+
+use Doctrine\Common\Collections\Collection;
+use Doctrine\ORM\Mapping as ORM;
+use Rector\Doctrine\Tests\CodeQuality\Rector\Class_\AddReturnDocBlockToCollectionPropertyGetterByToManyAnnotationRector\Source\Training;
+
+#[ORM\Entity()]
+class Trainer
+{
+    /**
+     * @var Collection|Training[]
+     */
+    #[ORM\OneToMany(targetEntity: Training::class, indexBy: "id", mappedBy: "trainer")]
+    private $trainings = [];
+
+    public function getTrainings()
+    {
+        return $this->trainings;
+    }
+
+    public function setTrainings($trainings)
+    {
+        $this->trainings = $trainings;
+    }
+}
+
+?>
+-----
+<?php
+
+namespace Rector\Doctrine\Tests\CodeQuality\Rector\Class_\AddReturnDocBlockToCollectionPropertyGetterByToManyAnnotationRector\Fixture\OneToMany;
+
+use Doctrine\Common\Collections\Collection;
+use Doctrine\ORM\Mapping as ORM;
+use Rector\Doctrine\Tests\CodeQuality\Rector\Class_\AddReturnDocBlockToCollectionPropertyGetterByToManyAnnotationRector\Source\Training;
+
+#[ORM\Entity()]
+class Trainer
+{
+    /**
+     * @var Collection|Training[]
+     */
+    #[ORM\OneToMany(targetEntity: Training::class, indexBy: "id", mappedBy: "trainer")]
+    private $trainings = [];
+
+    /**
+     * @return \Doctrine\Common\Collections\Collection<\Rector\Doctrine\Tests\CodeQuality\Rector\Class_\AddReturnDocBlockToCollectionPropertyGetterByToManyAnnotationRector\Source\Training>
```

Comment:
Both values have to filled `Collection<key, value>`, as first is key, next is value, see:

https://github.com/doctrine/collections/blob/2790cb57e6510e4404873bdca6688e447550d720/src/Collection.php#L29-L30

Reply:
per @AlexeyKosov at issue https://github.com/rectorphp/rector/issues/8791#issue-2468347080

the key is not necessarily need to be defined, we use `\Doctrine\Common\Collections\Collection<ClassName>` too in our project, and working well.

Reply:
That doesn't respect the parent interface contract:

> should have the Collection<string, Training> type rather than Collection<int, Training>.

![Screenshot from 2024-12-21 09-49-00](https://github.com/user-attachments/assets/37f571d0-8162-4360-a07f-2318a56c216a)

Both key and value should be included.

Reply:
I think for win win solution, skip php doc change is better, since lookup target class defined by `indexBy` may or may not work.

I will create new PR

Reply:
see new PR:

- https://github.com/rectorphp/rector-doctrine/pull/361